pkg/render/egressgateway/egressgateway.go

```diff
@@ -0,0 +1,438 @@
+		VolumeSource: corev1.VolumeSource{CSI: &corev1.CSIVolumeSource{Driver: "csi.tigera.io"}},
+	}
+}
+
+func (c *component) egwVolumeMounts() []corev1.VolumeMount {
+	return []corev1.VolumeMount{
+		corev1.VolumeMount{Name: "policysync", MountPath: "/var/run/calico"},
+	}
+}
+
+func (c *component) egwEnvVars() []corev1.EnvVar {
+	icmpProbeIPs, icmpInterval, icmpTimeout := c.getICMPProbe()
+	httpProbeURLs, httpInterval, httpTimeout := c.getHTTPProbe()
+	egressPodIp := &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}}
+	return []corev1.EnvVar{
+		{Name: "HEALTH_TIMEOUT_DATASTORE", Value: c.getHealthTimeoutDs()},
+		{Name: "HEALTH_PORT", Value: fmt.Sprintf("%d", DefaultHealthPort)},
+		{Name: "ICMP_PROBE_IPS", Value: icmpProbeIPs},
+		{Name: "ICMP_PROBE_INTERVAL", Value: icmpInterval},
+		{Name: "ICMP_PROBE_TIMEOUT", Value: icmpTimeout},
+		{Name: "HTTP_PROBE_URLS", Value: httpProbeURLs},
+		{Name: "HTTP_PROBE_INTERVAL", Value: httpInterval},
+		{Name: "HTTP_PROBE_TIMEOUT", Value: httpTimeout},
+		{Name: "EGRESS_POD_IP", ValueFrom: egressPodIp},
+		{Name: "EGRESS_VXLAN_VNI", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanVNI)},
+		{Name: "LOG_SEVERITY", Value: c.config.EgressGW.GetLogSeverity()},
+	}
+}
+
+func (c *component) egwInitEnvVars() []corev1.EnvVar {
+	egressPodIp := &corev1.EnvVarSource{FieldRef: &corev1.ObjectFieldSelector{FieldPath: "status.podIP"}}
+	return []corev1.EnvVar{
+		{Name: "EGRESS_VXLAN_VNI", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanVNI)},
+		{Name: "EGRESS_VXLAN_PORT", Value: fmt.Sprintf("%d", c.config.EgressGWVxlanPort)},
+		{Name: "EGRESS_POD_IP", ValueFrom: egressPodIp},
+	}
+}
+
+func (c *component) egwPodSecurityPolicy() *policyv1beta1.PodSecurityPolicy {
+	boolTrue := true
+	namespacedName := fmt.Sprintf("%s-%s", c.config.EgressGW.Namespace, c.config.EgressGW.Name)
+	psp := podsecuritypolicy.NewBasePolicy()
+	psp.GetObjectMeta().SetName(namespacedName)
+	psp.Spec.AllowedCapabilities = []corev1.Capability{
+		corev1.Capability("NET_ADMIN"),
+	}
+	psp.Spec.AllowPrivilegeEscalation = &boolTrue
+	psp.Spec.HostIPC = true
+	psp.Spec.HostNetwork = true
+	psp.Spec.HostPID = true
+	psp.Spec.Privileged = true
+	psp.Spec.RunAsUser = policyv1beta1.RunAsUserStrategyOptions{
+		Rule: policyv1beta1.RunAsUserStrategyRunAsAny,
+	}
+	psp.Spec.SELinux = policyv1beta1.SELinuxStrategyOptions{
+		Rule: policyv1beta1.SELinuxStrategyRunAsAny,
+	}
+	psp.Spec.SupplementalGroups = policyv1beta1.SupplementalGroupsStrategyOptions{
+		Rule: policyv1beta1.SupplementalGroupsStrategyRunAsAny,
+	}
+	psp.Spec.FSGroup = policyv1beta1.FSGroupStrategyOptions{
+		Rule: policyv1beta1.FSGroupStrategyRunAsAny,
+	}
+	psp.Spec.Volumes = []policyv1beta1.FSType{
+		policyv1beta1.CSI,
+		policyv1beta1.Projected,
+	}
+	return psp
+}
+
+func (c *component) egwClusterRole() *rbacv1.ClusterRole {
+	namespacedName := fmt.Sprintf("%s-%s", c.config.EgressGW.Namespace, c.config.EgressGW.Name)
+	rules := []rbacv1.PolicyRule{
```

Comment:
Should only include these rules if `usePSP` is enabled

Reply:
```egwClusterRole``` is called only when ```usePSP``` is Enabled.

Reply:
Then we need to also delete them when usePSP is Disabled, otherwise these configs will stay around when a user upgrades their cluster to a version of k8s that doesn't have PSP support.